### PR TITLE
feat(ila): update content pack

### PR DESCRIPTION
**Pull request by tenthirtyam**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

Add support to update content packs in vRealize Log Insight from the in-product content pack marketplace.

- Updates `Install-vRLIContentPack`
- Updates `Get-vRLIMarketplaceMetadata`
- Adds `Update-vRLIContentPack`
- Refactors `Enable-vRLIContentPack` to support a broader set of content packs.

Signed-off-by: Ryan Johnson <johnsonryan@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [x] This is a refactoring update.
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes: #56

**Test and Documentation Coverage**

- [ ] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -11016,6 +11030,89 @@ Function Enable-vRLIContentPack {
 }
 Export-ModuleMember -Function Enable-vRLIContentPack
 
+Function Update-vRLIContentPack {
+    <#
+        .SYNOPSIS
+        Updates the vRealize Log Insight content pack from the marketplace.
+
+        .DESCRIPTION
+        The Update-vRLIContentPack cmdlet installs a designated vRealize Log Insight content pack from the online
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:11039]: Should 'cmdlet installs a designated'
Be 'cmdlet updates a designated'

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -11016,6 +11030,89 @@ Function Enable-vRLIContentPack {
...
+                                        if ($contentPackStatus = (Get-vRLIContentPack | Where-Object { $_.name -eq $contentPackName }) | Out-Null) {
+                                            Write-Output "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)): SUCCESSFUL"
+                                        } else {
+                                            Write-Error "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)): POST_VALIDATION_FAILED"
+                                        }   
+                                    } else {
+                                        Write-Error "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)), no update available: PRE_VALIDATION_FAILED"
+                                    }
+                                } else {
+                                    Write-Warning "Retrieving content pack ($contentPackFile) metadata from the marketplace: PRE_VALIDATION_FAILED"
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:11100]: Should be Write-Error ?

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -11016,6 +11030,89 @@ Function Enable-vRLIContentPack {
...
+                                }
+                            } else {
+                                Write-Warning "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)), not installed: PRE_VALIDATION_FAILED"
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:11103]: Should be 'Write-Error'

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -30235,16 +30332,25 @@ Function Get-vRLIMarketplaceMetadata {
...
-        $ghHeaders = New-Object 'System.Collections.Generic.Dictionary[[String],[String]]'
-        $ghHeaders.Add('Accept', 'application/vnd.github.VERSION.raw')
-        $ghHeaders.Add('Authorization', "Basic $ghToken")
-
-        # Get the content pack metadata from the GitHub repository
-        $uri = 'https://api.github.com/repos/vmw-loginsight/vlcp/contents/content/'
+            $ghToken = 'bGktbWFya2V0cGxhY2U6Y2UzYjBkODFjYzczMTJhZjk5ZDYzMjFjZDlkMTUzOTc4ZjZlZjM2NQ=='
+            $ghHeaders = New-Object 'System.Collections.Generic.Dictionary[[String],[String]]'
+            $ghHeaders.Add('Accept', 'application/vnd.github.VERSION.raw')
+            $ghHeaders.Add('Authorization', "Basic $ghToken")
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:30345]: Are these really tabbed in ?

**Review by GaryJBlake — CHANGES_REQUESTED** [review]
Nice enhancement :)

File: PowerValidatedSolutions.psm1 (hunk 1 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:11039]: Addressed in amended commit.

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -11016,6 +11030,89 @@ Function Enable-vRLIContentPack {
...
+                                    $json = $response | ConvertTo-Json -Depth 100 -Compress
+                                    if ($response.contentVersion -lt $contentPackVersion) {
+                                        Install-vRLIContentPack -update -json $json | Out-Null
+                                        if ($contentPackStatus = (Get-vRLIContentPack | Where-Object { $_.name -eq $contentPackName }) | Out-Null) {
+                                            Write-Output "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)): SUCCESSFUL"
+                                        } else {
+                                            Write-Error "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)): POST_VALIDATION_FAILED"
+                                        }   
+                                    } else {
+                                        Write-Error "Updating content pack ($contentPackName) on vRealize Log Insight ($($vcfVrliDetails.fqdn)), no update available: PRE_VALIDATION_FAILED"
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:11097]: Addressed in amended commit.

File: PowerValidatedSolutions.psm1 (hunk 3 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:11103]: Addressed in amended commit.

File: PowerValidatedSolutions.psm1 (hunk 2 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:11100]: Addressed in amended commit.

File: PowerValidatedSolutions.psm1 (hunk 4 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1]: Addressed in amended commit.

**Comment by tenthirtyam** [comment]
I've kept this pull request in draft status until I run through some final tests.

**Review by GaryJBlake — APPROVED** [review]

**Pull request merged**